server/issue.go

```diff
@@ -511,55 +477,41 @@ func (s *Server) getPipelineCreateForDatabaseCreate(ctx context.Context, issueCr
 			return nil, fmt.Errorf("failed to create restore database task, unable to marshal payload %w", err)
 		}
 
+		taskIndexDAGList = append(taskIndexDAGList, api.TaskIndexDAG{
```

Comment:
Let's explicitly express the task dependency as `{FromIndex: 0, ToIndex: 1}`.

Do not need to make it **looks** generic for now, which adds mind burden.